app/controllers/job_invocations_controller.rb

```diff
@@ -7,7 +7,7 @@ class JobInvocationsController < ApplicationController
   def new
     return @composer = prepare_composer if params[:feature].present?
     ui_params = {
-      :host_ids => params[:host_ids],
+      :host_ids => params.key?(:search) ? Host.search_for(params[:search]).pluck(:id) : params[:host_ids],
```

Comment:
I think more natural and effective would be just using the search query directly inside the targeting, instead of expanding it here.

```patch
diff --git a/app/controllers/job_invocations_controller.rb b/app/controllers/job_invocations_controller.rb
index 8b78261..5ef03f7 100644
--- a/app/controllers/job_invocations_controller.rb
+++ b/app/controllers/job_invocations_controller.rb
@@ -7,8 +7,9 @@ class JobInvocationsController < ApplicationController
   def new
     return @composer = prepare_composer if params[:feature].present?
     ui_params = {
-      :host_ids => params.key?(:search) ? Host.search_for(params[:search]).pluck(:id) : params[:host_ids],
+      :host_ids => params[:host_ids],
       :targeting => {
+        :search_query => params[:search],
         :targeting_type => Targeting::STATIC_TYPE,
         :bookmark_id => params[:bookmark_id]
       }
```